File: src/FSharpVSPowerTools/XmlDocCommandFilterProvider.cs
using System;
using System.Diagnostics;
using System.ComponentModel.Composition;
using System.Runtime.InteropServices;
using Microsoft.VisualStudio;
using Microsoft.VisualStudio.Shell;
using Microsoft.VisualStudio.Editor;
using Microsoft.VisualStudio.OLE.Interop;
using Microsoft.VisualStudio.Text.Editor;
using Microsoft.VisualStudio.TextManager.Interop;
using Microsoft.VisualStudio.Utilities;
using Microsoft.VisualStudio.Text;
using FSharpVSPowerTools.XmlDoc;
using FSharpVSPowerTools.ProjectSystem;

// Useful reference: http://msdn.microsoft.com/en-us/library/dd885243.aspx
namespace FSharpVSPowerTools
{
    [Export(typeof(IVsTextViewCreationListener))]
    [ContentType("F#")]
    [TextViewRole(PredefinedTextViewRoles.Interactive)]
    public class XmlDocCommandFilterProvider : IVsTextViewCreationListener
    {
        [Import]
        private ITextDocumentFactoryService textDocumentFactoryService = null;

        [Import]
        private IVsEditorAdaptersFactoryService editorFactory = null;

        [Import]
        private VSLanguageService fsharpVsLanguageService = null;

        [Import(typeof(SVsServiceProvider))]
        private System.IServiceProvider serviceProvider = null;

        public void VsTextViewCreated(IVsTextView textViewAdapter)
        {
            var wpfTextView = editorFactory.GetWpfTextView(textViewAdapter);
            if (wpfTextView == null) return;

            var generalOptions = Utils.GetGeneralOptionsPage(serviceProvider);
            if (!generalOptions.XmlDocEnabled) return;

            ITextDocument doc;
            if (textDocumentFactoryService.TryGetTextDocument(wpfTextView.TextBuffer, out doc))
            {
                new XmlDocFilter(textViewAdapter, wpfTextView, doc.FilePath, fsharpVsLanguageService);
            }
        }
    }
}

File: src/FSharpVSPowerTools.Pure/Commands/XmlDocCommandFilterProvider.fs
module FSharpVSPowerTools.XmlDocCommandFilterProvider

open System.Diagnostics                         
open Microsoft.VisualStudio                     
open Microsoft.VisualStudio.Shell               
open Microsoft.VisualStudio.Editor              
open Microsoft.VisualStudio.OLE.Interop         
open Microsoft.VisualStudio.Text.Editor         
open Microsoft.VisualStudio.TextManager.Interop 
open Microsoft.VisualStudio.Utilities           
open Microsoft.VisualStudio.Text                
open System                                     
open System.ComponentModel.Composition          
open System.Runtime.InteropServices             
open FSharpVSPowerTools.XmlDoc                  
open FSharpVSPowerTools.ProjectSystem           

// Useful reference: http://msdn.microsoft.com/en-us/library/dd885243.aspx

[<Export (typeof<IVsTextViewCreationListener>)>]
[<ContentType "F#" >]
[<TextViewRole (PredefinedTextViewRoles.Interactive)>]
type XmlDocCommandFilterProvider [<ImportingConstructor>]
    ( [<Import(typeof<SVsServiceProvider>)>] 
    serviceProvider                 :   IServiceProvider                ,
    textDocumentFactoryService      :   ITextDocumentFactoryService     ,
    editorFactory                   :   IVsEditorAdaptersFactoryService ,    
    projectFactory                  :   ProjectFactory                  ,
    vsLanguageService               :   VSLanguageService               ) =   

    interface IVsTextViewCreationListener with
        member __.VsTextViewCreated textViewAdapter = 
            unitMaybe {
                let! generalOptions = Setting.tryGetGeneralOptions serviceProvider
                let! wpfTextView = editorFactory.TryGetWpfTextView textViewAdapter
                let! doc = textDocumentFactoryService.TryDocumentFromBuffer wpfTextView.TextBuffer  
                if generalOptions.XmlDocEnabled then 
                    XmlDocFilter( textViewAdapter, wpfTextView, doc.FilePath, projectFactory, 
                        vsLanguageService,OpenDocumentsTracker textDocumentFactoryService, serviceProvider)
                    |> ignore
            } 


         

